var/spack/repos/builtin/packages/ceed/package.py

```diff
@@ -43,9 +39,13 @@ class Ceed(BundlePackage):
     # LibCEED
     # ceed 4.0
     depends_on('libceed@0.8~cuda', when='@4.0.0~cuda')
-    depends_on('libceed@0.8+cuda+magma', when='@4.0.0+cuda')
-    depends_on('libceed@0.8~hip', when='@4.0.0~hip')
-    depends_on('libceed@0.8+hip', when='@4.0.0+hip')
+    for arch in CudaPackage.cuda_arch_values:
+        depends_on('libceed@0.8+cuda+magma cuda_arch={0}'.format(arch),
+                   when='@4.0.0+cuda cuda_arch={0}'.format(arch))
+    depends_on('libceed@0.8~rocm', when='@4.0.0~rocm')
+    for target in ROCmPackage.amdgpu_targets:
+        depends_on('libceed@0.8+rocm amdgpu_target={0}'.format(target),
+                   when='@4.0.0+rocm amdgpu_target={0}'.format(target))
```

Comment:
@becker33 Is this what you meant when you said (on Slack) that the variant needs to be forwarded to each dependent package? The spec makes sense with this change, though it's tedious going and all the BundlePackages would need this treatment.